blockchain/blockchain.go

```diff
@@ -854,6 +859,55 @@ func (b *Blockchain) revertHead(txn db.Transaction) error {
 	return txn.Set(db.ChainHeight.Key(), heightBin)
 }
 
+func (b *Blockchain) storeEmptyPending(txn db.Transaction, latestHeader *core.Header) error {
+	pendingBlock := &core.Block{
+		Header: &core.Header{
+			Hash:             nil, // Set below.
+			ParentHash:       latestHeader.Hash,
+			Number:           latestHeader.Number + 1,
+			GlobalStateRoot:  latestHeader.GlobalStateRoot,
+			SequencerAddress: latestHeader.SequencerAddress,
+			TransactionCount: 0,
+			EventCount:       0,
+			Timestamp:        latestHeader.Timestamp + 1,
+			ProtocolVersion:  supportedStarknetVersion.String(),
+			ExtraData:        nil,
+			EventsBloom:      nil,
+			GasPrice:         new(felt.Felt),
+		},
+		Transactions: make([]core.Transaction, 0),
+		Receipts:     make([]*core.TransactionReceipt, 0),
+	}
+	var err error
+	pendingBlock.Hash, _, err = core.BlockHash(pendingBlock, b.network, nil)
+	if err != nil {
+		return err
+	}
+
+	emptyPending := &Pending{
+		Block: pendingBlock,
+		StateUpdate: &core.StateUpdate{
+			BlockHash: pendingBlock.Hash,
+			NewRoot:   pendingBlock.GlobalStateRoot,
+			OldRoot:   latestHeader.GlobalStateRoot,
+			StateDiff: &core.StateDiff{
+				StorageDiffs:      make(map[felt.Felt][]core.StorageDiff, 0),
+				Nonces:            make(map[felt.Felt]*felt.Felt, 0),
+				DeployedContracts: make([]core.DeployedContract, 0),
+				DeclaredV0Classes: make([]*felt.Felt, 0),
+				DeclaredV1Classes: make([]core.DeclaredV1Class, 0),
+				ReplacedClasses:   make([]core.ReplacedClass, 0),
+			},
+		},
+		NewClasses: make(map[felt.Felt]core.Class, 0),
+	}
+	emptyPendingBytes, err := encoder.Marshal(emptyPending)
+	if err != nil {
+		return err
+	}
+	return txn.Set(db.Pending.Key(), emptyPendingBytes)
```

Comment:
can we use `StorePending` here?

Reply:
We could, but it will perform some useless verification. Is that acceptable?

Reply:
> but it will perform some useless verification

Let's think of them as sanity checks :D

Reply:
Actually, we can't use `StorePending` as-is since it creates a separate transaction.

We can create function `storePending` that is called by `storeEmptyPending` and `StorePending`. Is that worth it? I don't mind either way.

Reply:
Let's create a `storePending(txn)` if you dont mind